prov/util/src/util_wait.c

```diff
@@ -424,3 +428,219 @@ int ofi_wait_fd_open(struct fid_fabric *fabric_fid, struct fi_wait_attr *attr,
 	free(wait);
 	return ret;
 }
+
+static void util_wait_yield_signal(struct util_wait *util_wait)
+{
+	struct util_wait_yield *wait_yield;
+
+	wait_yield = container_of(util_wait, struct util_wait_yield, util_wait);
+
+	fastlock_acquire(&wait_yield->signal_lock);
+	wait_yield->signal = 1;
+	fastlock_release(&wait_yield->signal_lock);
+}
+
+static int util_wait_yield_run(struct fid_wait *wait_fid, int timeout)
+{
+	struct util_wait_yield *wait = container_of(wait_fid,
+			struct util_wait_yield, util_wait.wait_fid);
+	struct ofi_wait_fid_entry *fid_entry;
+	int ret = 0;
+
+	fastlock_acquire(&wait->signal_lock);
+	if (wait->signal) {
+		wait->signal = 0;
+		fastlock_release(&wait->signal_lock);
+		return FI_SUCCESS;
+	}
+	wait->signal = 0;
+	fastlock_release(&wait->signal_lock);
```

Comment:
I don't know that we need the above code at all.  signal is set to 0 initially.  It's reset to 0 every time we exit this call.  If we ever enter this call with it set to 1, we'll skip the while loop below (once it's fixed), reset signal to 0 and return success.